ocaml/xapi/xapi.ml

```diff
@@ -557,7 +558,16 @@ let check_network_reset () =
            )) in
          match pifs with
          | [] -> error "management PIF %s not found" device
-         | pif :: _ ->
+         | phy_pif :: _ ->
+           let pif =
+             match vlan with
+             | Some vlan ->
```

Comment:
I would consider creating a function to create a network in order to simplify the code here:

```
match pifs, vlan with
| []    ,  _         -> error "management PIF %s not found" device
| pif::_,  None      -> pif
| pif::_,  Some vlan -> create_network pif vlan
```